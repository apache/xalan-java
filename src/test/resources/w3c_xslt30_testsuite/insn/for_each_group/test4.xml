<cities>
  <city name="milan"  country="italy"   pop="5"/>
  <city name="paris"  country="france"  pop="7"/>
  <city name="munich" country="germany" pop="4"/>
  <city name="lyon"   country="france"  pop="2"/>
  <city name="london"  country="england"   pop="10"/>
  <city name="bristol"   country="england"  pop="5.0"/>
  <city name="manchester"  country="england"  pop="7.0"/>
  <city name="liverpool" country="england" pop="7"/>
  <city name="sheffield"   country="england"  pop="05"/>
  <city name="venice" country="italy"   pop="1"/>  
</cities>