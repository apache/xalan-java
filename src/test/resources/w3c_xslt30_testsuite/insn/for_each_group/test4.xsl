<?spec xslt#grouping?>
<xsl:transform xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
               xmlns:xs="http://www.w3.org/2001/XMLSchema"               
               exclude-result-prefixes="xs"
			   version="3.0">
			   
  <!-- use with test4.xml -->
  
  <!-- An XSL test case, to verify fix for W3C XSLT 3.0 xsl:for-each-group 
       test case for-each-group-043.
       (XSLT 3.0 grouping using group-by with composite keys)   
   -->			   
  
  <xsl:output method="xml" indent="yes"/>
  
  <xsl:template match="/">
     <out>
	     <xsl:for-each-group select="/*/city" 
	                         group-by="@country, xs:decimal(@pop)"
	                         composite="yes">
	        <group country="{current-grouping-key()[1]}" pop="{current-grouping-key()[2]}">
	           <xsl:copy-of select="current-group()"/>
	        </group>
	     </xsl:for-each-group>
      </out>
   </xsl:template>
	
</xsl:transform>
