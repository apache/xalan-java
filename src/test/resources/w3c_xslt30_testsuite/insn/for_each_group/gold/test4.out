<?xml version="1.0" encoding="UTF-8"?><out>
  <group pop="5" country="italy">
    <city country="italy" name="milan" pop="5"/>
  </group>
  <group pop="7" country="france">
    <city country="france" name="paris" pop="7"/>
  </group>
  <group pop="4" country="germany">
    <city country="germany" name="munich" pop="4"/>
  </group>
  <group pop="2" country="france">
    <city country="france" name="lyon" pop="2"/>
  </group>
  <group pop="10" country="england">
    <city country="england" name="london" pop="10"/>
  </group>
  <group pop="5" country="england">
    <city country="england" name="bristol" pop="5.0"/>
    <city country="england" name="sheffield" pop="05"/>
  </group>
  <group pop="7" country="england">
    <city country="england" name="manchester" pop="7.0"/>
    <city country="england" name="liverpool" pop="7"/>
  </group>
  <group pop="1" country="italy">
    <city country="italy" name="venice" pop="1"/>
  </group>
</out>
